### ISSUE TITLE
Wiki development

### ISSUE THREAD

**Issue by thegr8dudfG** [sscarduzio/elasticsearch-readonlyrest-plugin#188]
Hi, can you please write another wiki page for development with a gradle project

thank you so much

**Comment by sscarduzio** [comment]
Hi @thegr8dudfG, the approach I'm using since we had to switch to ES 5.x + Gradle is not as nice as the Maven one: no more in-IDE run.
I do  
```bash
$ bin/build.sh && bin/test.sh some/adhoc/elasticsearch.yml 
```
Then I wait a minute the docker container is created and launched.
Then I just **remote debug** with intelliJ on port 8000.

I know, quite a drag waiting for docker every time. I'm sure there's room for improvement on the in-IDE debug dev cycle, like using docker volumes to load the readonlyrest.jar and elasticsearch.yml (which now are copied over in `docker build` phase inside `bin/test.sh`) but never got around to do it.

**Comment by sscarduzio** [comment]
I'm not sure how @coutoPL and @innotech-research handle this, maybe they don't even use the debugger (I'm a very heavy debugger user, especially in this project where sadly everything is runtime injected by guice and reflection is a something we face every day). I know many people just print logs and it's enough for them.

**Comment by coutoPL** [comment]
I've run project in intellij. It can be done in few simple steps:
1. import as a gradle project
2. project structure -> modules -> readonlyrest_main -> change scope of `provided` dependencies to `compile` -> save
3. edit configurations -> add -> application:
- main class: `org.elasticsearch.bootstrap.Elasticsearch`
- VM options: `-Djava.security.policy=/Users/couto/Workspace/github/elasticsearch-readonlyrest-plugin/src/test/eshome/plugins/readonlyrest/plugin-security.policy -Djava.awt.headless=true -Des.insecure.allow.root=true -Dfile.encoding=UTF-8 -Djna.nosys=true -Des.path.home=/Users/couto/Workspace/github/elasticsearch-readonlyrest-plugin/src/test/eshome -Dlog4j2.disable.jmx=true -Xms2g -Xmx2g -Dlog4j2.disable.jmx=true` (paths should be changed ... maybe relative paths also work)
- use classpath of module: `readonlyrest_main`
4. I had to add `transport.type: local` to `src/test/eshome/config/elasticsearch.yml`
5. Run/Debug ... ES should be visible at `http://localhost:9200/`

**Comment by innotech-research** [comment]
ElasticSearch plugin development with debuggers is a nightmare. I don't think I've ever managed it.
I developed a plugin for ES 2.3 that did everything I needed. I did this by looking through elastic source code, and a lot of logging statements. The bit I found most difficult was plugging into elasticsearch itself. Once I'd mastered that then doing what I wanted was relatively straightforward.
I then looked at porting to ES5 and baulked at the prospect when I saw what I was up against. I found this project which had done the injecting work already and has a pretty good build framework. I build the code, copy it to an elastic node, shut it down, remove and install the plugin, start it up, and run a load of curl scripts.
